spec/account_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'bankaccounts'
+require 'pry'
+require 'spec_helper'
+
+describe BankAccount::Account do
+
+  describe "It is initialized with assignments" do
+    let (:bankaccount) {BankAccount::Account.new(1, 10000)}
+
+    it "has an ID of 1" do
+      expect(bankaccount.id).to eq(1)
+    end
+
+    it "has an initial balance of 10000" do
+      expect(bankaccount.initial_balance).to eq(10000)
+    end
+
+    it "raises an ArgumentError if a negative account balance is initialized" do
+      expect{BankAccount::Account.new(3, -5000)}.to raise_error(ArgumentError)
+    end
+  end
+
+  describe "It has instance methods with functionality" do
+    let (:bankaccount) {BankAccount::Account.new(2, 10000)}
+
+    it "Returns a balance of 5000 when 5000 is withdrawn" do
+      expect(bankaccount.withdraw(5000)).to eq(5000)
```

Comment:
This is very much a personal preference, but I think it's worth duplicating the: 
`bankaccount = BankAccount::Account.new(2, 10000)`
line here, rather than in the `let` as it makes it much easier to see the math that is being done. Otherwise figuring out that the math being done is 10000 - 5000 = 5000 is split by several lines of code.